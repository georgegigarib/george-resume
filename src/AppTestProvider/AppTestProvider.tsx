import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
