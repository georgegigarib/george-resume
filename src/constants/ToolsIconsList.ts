import JsIcon from '@/assets/icons/tools/Javascript.svg'

export type ToolInfo = {
  name: string
  icon: IconName
  color: string
  bgColor: string
  darkBgColor?: string
  borderColor: string
  darkBorderColor?: string
  darkColor?: string
  svg?: string
  custom?: boolean
}

export type ToolIconProps = {
  tool: IconName
  size?: '5x' | '4x' | '3x' | '2x' | 'xl' | 'lg' | 'md' | 'sm' | 'xs'
}

export enum IconName {
  React = 'react',
  ReactNative = 'reactnative',
  TailwindCSS = 'tailwindcss',
  JavaScript = 'js',
  TypeScript = 'ts-circle',
  VueJs = 'vuejs',
  NextJS = 'nextjs',
  NodeJS = 'nodejs',
  HTML = 'html',
  CSS = 'css',
  Express = 'express',
  Spotify = 'spotify',
  Git = 'git',
  Redis = 'redis',
  Bun = 'bun',
  Docker = 'docker',
  Pinia = 'pinia',
  Sass = 'sass',
  Bootstrap = 'bootstrap',
  Angular = 'angular',
  C = 'c',
  CSharp = 'csharp',
  Django = 'django',
  FramerMotion = 'framer-motion',
  Go = 'go',
  Graphql = 'graphql',
  Javalanguage = 'java',
  Laravel = 'laravel',
  Mui = 'mui',
  Jest = 'jest',
  Vitest = 'vitest',
  Cypress = 'cypress',
  MySQL = 'mysql',
  MongoDb = 'mongodb',
  Php = 'php',
  Postgresql = 'postgresql',
  Python = 'python',
  Redux = 'redux',
  Zustand = 'zustand',
  Svelte = 'svelte',
}

export const toolsIconsList: ToolInfo[] = [
  {
    name: 'React',
    icon: IconName.React,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'React',
    icon: IconName.ReactNative,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Tailwind CSS',
    icon: IconName.TailwindCSS,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#38BDF8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'JavaScript',
    icon: IconName.JavaScript,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
    custom: true,
    svg: JsIcon,
  },
  {
    name: 'Jest',
    icon: IconName.Jest,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MySQL',
    icon: IconName.MySQL,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#38BDF8',
  },
  {
    name: 'Vitest',
    icon: IconName.Vitest,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
  },
  {
    name: 'Cypress',
    icon: IconName.Cypress,
    color: '#000000FF',
    bgColor: '#B4B4B4FF',
    darkBgColor: '#BABABAFF',
    borderColor: '#C7B008FF',
  },
  {
    name: 'TypeScript',
    icon: IconName.TypeScript,
    color: '#3178C6',
    bgColor: '#FAFAFAFF',
    borderColor: '#3178C6',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Next.js',
    icon: IconName.NextJS,
    color: '#0000000',
    bgColor: '#FAFAFAFF',
    borderColor: '#555555FF',
    darkBgColor: '#3E3E3EFF',
    darkColor: '#FFFFFF',
    darkBorderColor: '#BFBFBFFF',
  },
  {
    name: 'Node.js',
    icon: IconName.NodeJS,
    color: '#539E43',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#539E43',
  },
  {
    name: 'HTML',
    icon: IconName.HTML,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'CSS',
    icon: IconName.CSS,
    color: '#264DE4',
    bgColor: '#FAFAFAFF',
    borderColor: '#264DE4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Express',
    icon: IconName.Express,
    color: '#333333',
    bgColor: '#FAFAFAFF',
    borderColor: '#CCCCCC',
  },
  {
    name: 'Spotify',
    icon: IconName.Spotify,
    color: '#1DB954',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#1DB954',
  },
  { name: 'Git', icon: IconName.Git, color: '#F05032', bgColor: '#F5F5F5', borderColor: '#F05032' },
  {
    name: 'Redis',
    icon: IconName.Redis,
    color: '#D82C20',
    bgColor: '#FAFAFAFF',
    borderColor: '#D82C20',
  },
  {
    name: 'Bun',
    icon: IconName.Bun,
    darkColor: '#ffffff',
    color: '#333333',
    bgColor: '#FAFAFAFF',
    darkBorderColor: '#bababa',
    borderColor: '#686767FF',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Docker',
    icon: IconName.Docker,
    color: '#2496ED',
    bgColor: '#FAFAFAFF',
    borderColor: '#2496ED',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Pinia',
    icon: IconName.Pinia,
    color: '#F0C723FF',
    darkBgColor: '#2C2C2C',
    bgColor: '#FAFAFAFF',
    borderColor: '#FFD939',
  },
  {
    name: 'Sass',
    icon: IconName.Sass,
    color: '#CC6699',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#CC6699',
  },
  {
    name: 'Bootstrap',
    icon: IconName.Bootstrap,
    color: '#A877F3FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#7952B3',
  },
  {
    name: 'Angular',
    icon: IconName.Angular,
    color: '#DD0031',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#DD0031',
  },
  {
    name: 'C',
    icon: IconName.C,
    color: '#00599C',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#00599C',
  },
  {
    name: 'C#',
    icon: IconName.CSharp,
    color: '#239120',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#239120',
  },
  {
    name: 'Django',
    icon: IconName.Django,
    color: '#092E20',
    bgColor: '#FAFAFAFF',
    borderColor: '#092E20',
  },
  {
    name: 'Framer Motion',
    icon: IconName.FramerMotion,
    color: '#EC5990',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#EC5990',
  },
  {
    name: 'Go',
    icon: IconName.Go,
    color: '#00ADD8',
    bgColor: '#FAFAFAFF',
    borderColor: '#00ADD8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'GraphQL',
    icon: IconName.Graphql,
    color: '#E10098',
    bgColor: '#FAFAFAFF',
    borderColor: '#E10098',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'java',
    icon: IconName.Javalanguage,
    color: '#007396',
    bgColor: '#FAFAFAFF',
    borderColor: '#007396',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Laravel',
    icon: IconName.Laravel,
    color: '#FF2D20',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF2D20',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Material UI',
    icon: IconName.Mui,
    color: '#0081CB',
    bgColor: '#FAFAFAFF',
    borderColor: '#0081CB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MongoDB',
    icon: IconName.MongoDb,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Vue Js',
    icon: IconName.VueJs,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PHP',
    icon: IconName.Php,
    color: '#959AE1FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#777BB4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PostgreSQL',
    icon: IconName.Postgresql,
    color: '#336791',
    bgColor: '#FAFAFAFF',
    borderColor: '#336791',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Python',
    icon: IconName.Python,
    color: '#3776AB',
    bgColor: '#FAFAFAFF',
    borderColor: '#3776AB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Redux',
    icon: IconName.Redux,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Zustand',
    icon: IconName.Zustand,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Svelte',
    icon: IconName.Svelte,
    color: '#FF3E00',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF3E00',
    darkBgColor: '#2C2C2C',
  },
]
