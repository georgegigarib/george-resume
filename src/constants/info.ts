export const WORK_EMAIL = 'georgegil.work@gmail.com'
export const SOCIAL_LINKS = {
  github: 'https://github.com/georgegigarib',
  linkedin: 'https://www.linkedin.com/in/georgegg/',
  instagram: 'URL_ADDRESS.com/GeorgeGil',
}
