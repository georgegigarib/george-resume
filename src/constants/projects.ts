import image1 from '@/assets/images/projects/movie-app.gif'
import image4 from '@/assets/images/projects/desktop-app-1.png'
import image6 from '@/assets/images/projects/desktop-app-2.png'
import { IconName } from '@/constants/ToolsIconsList'

export type ProjectInfoType = {
  src: string
  title: string
  description: string
  url: string
  tools: IconName[]
}

export const mobileProjects: ProjectInfoType[] = [
  {
    src: image1,
    title: 'dev.projects.projects.mobile.project1.title',
    description: 'dev.projects.projects.mobile.project1.description',
    url: 'https://movie-app.example.com',
    tools: [
      IconName.React,
      IconName.TailwindCSS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git,
    ],
  },
  {
    src: image1,
    title: 'dev.projects.projects.mobile.project2.title',
    description: 'dev.projects.projects.mobile.project2.description',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS, IconName.JavaScript],
  },
  {
    src: image1,
    title: 'dev.projects.projects.mobile.project3.title',
    description: 'dev.projects.projects.mobile.project3.description',
    url: 'https://project2.example.com',
    tools: [
      IconName.React,
      IconName.TailwindCSS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git,
    ],
  },
]

export const desktopProjects: ProjectInfoType[] = [
  {
    src: image6,
    title: 'dev.projects.projects.desktop.project1.title',
    description: 'dev.projects.projects.desktop.project1.description',
    url: 'https://desktop1.example.com',
    tools: [
      IconName.React,
      IconName.TailwindCSS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git,
    ],
  },
  {
    src: image4,
    title: 'dev.projects.projects.desktop.project2.title',
    description: 'dev.projects.projects.desktop.project2.description',
    url: 'https://desktop2.example.com',
    tools: [
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git,
    ],
  },
  {
    src: image4,
    title: 'dev.projects.projects.desktop.project3.title',
    description: 'dev.projects.projects.desktop.project3.description',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git],
  },
  {
    src: image6,
    title: 'dev.projects.projects.desktop.project4.title',
    description: 'dev.projects.projects.desktop.project4.description',
    url: 'https://desktop2.example.com',
    tools: [IconName.NextJS, IconName.NodeJS, IconName.TailwindCSS, IconName.Redis, IconName.Git],
  },
  {
    src: image4,
    title: 'dev.projects.projects.desktop.project5.title',
    description: 'dev.projects.projects.desktop.project5.description',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git],
  },
]
