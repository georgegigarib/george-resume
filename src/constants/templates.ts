import { IconName } from './ToolsIconsList'

export interface Template {
  tools: IconName[]
  url: string
  hint: string
  bgColor: string
  description: string
}

export const devTemplates: Template[] = [
  {
    tools: [IconName.TypeScript, IconName.React, IconName.TailwindCSS, IconName.Bun, IconName.CSS],
    url: 'https://github.com/example/fast-render-template',
    hint: 'fast rendering',
    bgColor: '#919469FF',
    description: 'A lightweight template designed for fast rendering with modern tools.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.NextJS,
      IconName.TailwindCSS,
      IconName.Docker,
      IconName.React
    ],
    url: 'https://github.com/example/next-ssr-template',
    hint: 'server rendering',
    bgColor: '#B99E4CFF',
    description: 'A server-side rendering starter template using Next.js and Docker.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.VueJs,
      IconName.TailwindCSS,
      IconName.Pinia,
      IconName.Docker
    ],
    url: 'https://github.com/example/vue-pinia-template',
    hint: 'vue simplicity',
    bgColor: '#6E8C9AFF',
    description: 'A clean Vue.js starter with Pinia and TailwindCSS for simplicity and scalability.'
  },
  {
    tools: [IconName.JavaScript, IconName.Angular, IconName.Sass, IconName.Docker, IconName.NodeJS],
    url: 'https://github.com/example/angular-enterprise-template',
    hint: 'enterprise apps',
    bgColor: '#A26B5AFF',
    description: 'A robust template for building enterprise-grade Angular applications.'
  },
  {
    tools: [IconName.TypeScript, IconName.React, IconName.Mui, IconName.Graphql, IconName.Django],
    url: 'https://github.com/example/react-graphql-template',
    hint: 'full stack',
    bgColor: '#877F95FF',
    description: 'Full-stack development made easy with React, MUI, and GraphQL backend.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.Svelte,
      IconName.TailwindCSS,
      IconName.Python,
      IconName.Postgresql
    ],
    url: 'https://github.com/example/svelte-minimal-template',
    hint: 'minimalist setup',
    bgColor: '#687969FF',
    description:
      'A minimalist yet powerful template using Svelte and Python with a PostgreSQL database.'
  },
  {
    tools: [IconName.JavaScript, IconName.React, IconName.Redux, IconName.Bootstrap, IconName.Php],
    url: 'https://github.com/example/ecommerce-template',
    hint: 'ecommerce base',
    bgColor: '#908C63FF',
    description: 'A ready-to-use base for building eCommerce solutions with React and PHP.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.React,
      IconName.Graphql,
      IconName.MongoDb,
      IconName.Docker
    ],
    url: 'https://github.com/example/data-intensive-template',
    hint: 'data intensive',
    bgColor: '#96659FFF',
    description: 'A powerful template for data-intensive applications using MongoDB and GraphQL.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.React,
      IconName.TailwindCSS,
      IconName.FramerMotion,
      IconName.NodeJS
    ],
    url: 'https://github.com/example/firebase-react-template',
    hint: 'real-time apps',
    bgColor: '#3E848DFF',
    description: 'A template for real-time apps with Firebase backend and React frontend.'
  }
]
