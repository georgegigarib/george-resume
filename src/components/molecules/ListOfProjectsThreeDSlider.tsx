import React, { useEffect, useState } from 'react'
import { useIsMobile } from '@/hooks/useIsMobile'
import initCarousel from '../../utils/carouselScript'
import { motion } from 'framer-motion'
import ProjectInfo from '@/components/atoms/ProjectInfo'
import ToggleProjectsViewButton from '@/components/atoms/ToggleProjectsViewButton'
import ThreeDImageSlider from '@/components/atoms/ThreeDImageSlider'
import { desktopProjects, mobileProjects, ProjectInfoType } from '@/constants/projects'
import clsx from 'clsx'

const MOBILE_SIZE_ON_MOBILE = { width: 115, height: 230 }
const MOBILE_SIZE_ON_DESKTOP = { width: 150, height: 280 }

const DESKTOP_SIZE_ON_MOBILE = { width: 190, height: 120 }
const DESKTOP_SIZE_ON_DESKTOP = { width: 260, height: 140 }

const CarouselComponent: React.FC = () => {
  const isMobile = useIsMobile()
  const [isDesktopMode, setIsDesktopMode] = useState(false)
  const projects = isDesktopMode ? desktopProjects : mobileProjects
  const [selectedProject, setSelectedProject] = useState<ProjectInfoType | null>(projects[0])

  const handleImageClick = (project: ProjectInfoType) => setSelectedProject(project)
  const toggleViewMode = () => setIsDesktopMode((prev) => !prev)

  const carouselSize = isDesktopMode
    ? isMobile
      ? DESKTOP_SIZE_ON_MOBILE
      : DESKTOP_SIZE_ON_DESKTOP
    : isMobile
      ? MOBILE_SIZE_ON_MOBILE
      : MOBILE_SIZE_ON_DESKTOP

  const carouselRadious = isDesktopMode ? (isMobile ? 30 : 40) : isMobile ? 19 : 24

  useEffect(() => {
    initCarousel(
      -200,
      carouselSize.width,
      carouselSize.height,
      true,
      projects.length * carouselRadious
    )
  }, [carouselSize, projects.length])

  return (
    <div
      className={clsx('mt-2 grid touch-none w-full', isMobile ? 'grid-rows-2' : 'grid-cols-2 p-4')}
    >
      <div className={isMobile ? '' : ''}>
        {selectedProject && (
          <ProjectInfo
            title={selectedProject.title}
            description={selectedProject.description}
            url={selectedProject.url}
            tools={selectedProject.tools}
          />
        )}
      </div>

      <ThreeDImageSlider
        projects={projects.map((project) => ({
          src: project.src,
          title: project.title,
          onClick: () => handleImageClick(project)
        }))}
        isDesktopMode={isDesktopMode}
      />

      <div className="absolute right-5 bottom-5">
        <ToggleProjectsViewButton isDesktopMode={isDesktopMode} onClick={toggleViewMode} />
      </div>

      <motion.div
        className={clsx(
          'dark:text-white d text-black absolute left-8 bottom-7 bg-slate-700 rounded-lg px-2 text-center md:text-start'
        )}
        initial={{ opacity: 0 }}
        animate={{ opacity: [0, 1, 1, 1, 1, 1, 1, 1, 0] }}
        transition={{
          times: [0, 0.5, 0.8, 1],
          duration: 6,
          ease: 'easeInOut',
          repeat: Infinity,
          repeatDelay: 1
        }}
      >
        <p>Don't be shy... spin it!</p>
      </motion.div>
    </div>
  )
}

export default CarouselComponent
