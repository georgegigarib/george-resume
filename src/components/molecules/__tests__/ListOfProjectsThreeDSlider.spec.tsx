import { render, screen, fireEvent } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import ListOfProjectsThreeDSlider from '@/components/molecules/ListOfProjectsThreeDSlider'
import { desktopProjects, mobileProjects } from '@/constants/projects'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}))

class ResizeObserver {
  observe() {}
  unobserve() {}
  disconnect() {}
}

global.ResizeObserver = ResizeObserver

vi.mock('@/constants/projects', () => ({
  desktopProjects: [
    {
      title: 'Desktop Project 1',
      description: 'Description for Desktop Project 1',
      url: 'http://example.com/desktop1',
      tools: ['React', 'TypeScript'],
      src: 'desktop1.png',
    },
  ],
  mobileProjects: [
    {
      title: 'Mobile Project 1',
      description: 'Description for Mobile Project 1',
      url: 'http://example.com/mobile1',
      tools: ['React Native', 'JavaScript'],
      src: 'mobile1.png',
    },
  ],
}))

// TODO: Fix this
describe('ListOfProjectsThreeDSlider Component', () => {
  const renderComponent = (platform: 'desktop' | 'mobile') =>
    render(
      <AppTestProvider>
        <ListOfProjectsThreeDSlider platform={platform} />
      </AppTestProvider>
    )

  it('renders correctly', () => {
    const { asFragment } = renderComponent('desktop')

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly with desktop projects', () => {
    renderComponent('desktop')

    desktopProjects.forEach((project) => {
      expect(screen.findByAltText(project.title)).toBeTruthy()
    })
  })

  it('renders correctly with mobile projects', () => {
    renderComponent('mobile')

    mobileProjects.forEach((project) => {
      expect(screen.findByAltText(project.title)).toBeTruthy()
    })
  })

  it('toggles between desktop and mobile view', () => {
    renderComponent('desktop')

    const toggleButton2 = screen.getByTestId('platform-toggle')
    fireEvent.click(toggleButton2)

    mobileProjects.forEach((project) => {
      expect(screen.findByAltText(project.title)).toBeTruthy()
    })
  })
})
