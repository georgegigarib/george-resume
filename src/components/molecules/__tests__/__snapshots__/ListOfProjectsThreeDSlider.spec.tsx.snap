// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListOfProjectsThreeDSlider Component > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mt-2 grid touch-none w-full grid-cols-2 p-4"
  >
    <div
      class=""
    >
      <div
        class=""
      >
        <span
          class="flex items-center"
        >
          <h2
            class="font-bold dark:text-white text-black"
          />
          <a
            class="ml-1"
            href="http://example.com/desktop1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1m2z4e8-MuiSvgIcon-root"
              data-testid="LaunchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
              />
            </svg>
          </a>
        </span>
        <div
          class="overflow-auto mt-2 mb-2 md:mb-4 max-h-20 md:max-h-[12vh] description-scrollbar"
        >
          <p
            class="dark:text-white text-black pr-8"
          />
        </div>
        <div
          class="mt-0"
        >
          <p
            class="dark:text-white font-bold text-black mb-2"
          />
          <div
            class="overflow-auto  md:hidden mt-2 max-h-10 h-14 flex flex-nowrap max-w-[65vw]"
          >
            <div
              class="rfm-marquee-container flex-wrap flex gap-2"
              style="--pause-on-hover: running; --pause-on-click: running; --width: 100%; --transform: none;"
            >
              <div
                class="rfm-marquee"
                style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite; --min-width: 100%;"
              >
                <div
                  class="rfm-initial-child-container"
                >
                  <div
                    class="rfm-child"
                    style="--transform: none;"
                  />
                  <div
                    class="rfm-child"
                    style="--transform: none;"
                  />
                </div>
              </div>
              <div
                class="rfm-marquee"
                style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite; --min-width: 100%;"
              >
                <div
                  class="rfm-child"
                  style="--transform: none;"
                />
                <div
                  class="rfm-child"
                  style="--transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="overflow-auto md:flex mt-2 max-h-20 md:max-h-[12vh] gap-1 description-scrollbar hidden flex-wrap"
          />
        </div>
      </div>
    </div>
    <div
      class=" top-[7vh]"
      id="drag-container"
    >
      <div
        id="spin-container"
        style="width: 260px; height: 140px; animation: spinRevert 200s infinite linear;"
      >
        <div
          class="absolute inset-0 flex items-center justify-center"
        >
          <span
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </span>
        </div>
        <img
          alt="Desktop Project 1"
          class="carousel-image opacity-0"
          src="desktop1.png"
          style="transform: rotateY(0deg) translateZ(40px); transition: transform 1s; transition-delay: 0.25s;"
        />
      </div>
      <div
        id="ground"
        style="width: 120px; height: 120px;"
      />
    </div>
    <div
      class="absolute right-5 bottom-5"
    >
      <button
        aria-label="Cambiar a vista Mobile"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wjzr1s-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-rlj5h7-MuiSvgIcon-root"
          data-testid="MobileScreenShareIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 1.01 7 1c-1.1 0-1.99.9-1.99 2v18c0 1.1.89 2 1.99 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-1.99-2-1.99M17 19H7V5h10zm-4.2-5.78v1.75l3.2-2.99L12.8 9v1.7c-3.11.43-4.35 2.56-4.8 4.7 1.11-1.5 2.58-2.18 4.8-2.18"
          />
        </svg>
      </button>
    </div>
    <div
      class="dark:text-white d text-black absolute left-8 bottom-7 bg-slate-700 rounded-lg px-2 text-center md:text-start"
      style="opacity: 0; will-change: opacity;"
    >
      <p />
    </div>
  </div>
</DocumentFragment>
`;
